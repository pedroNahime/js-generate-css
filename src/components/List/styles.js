import styled from 'styled-components'

export const Container = styled.div`
  flex: 1;
  margin-top: 25px;
  display: flex;
  flex-direction: column;
  overflow: auto;
`

export const Title = styled.h1`
  font-size: 48px;
`

export const Wrapper = styled.div`
  margin-top: 20px;
  display: flex;
  flex-wrap: wrap;
  align-items: center;
`
