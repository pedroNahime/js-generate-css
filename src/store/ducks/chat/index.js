import { createActions, createReducer } from 'reduxsauce'

export const { Types, Creators } = createActions({
  toggleChat: [],
  getChat: [],
  addMessage: ['message']
})

export const MoviesTypes = Types
export default Creators
/**
 * Handlers
 */
export const INITIAL_STATE = {
  active: false,
  messages: []
}

export const toggle = (state = INITIAL_STATE) => ({
  ...state,
  active: !state.active
})
export const get = (state = INITIAL_STATE) => ({
  ...state
})

export const addMessage = (state = INITIAL_STATE, action) => ({
  ...state,
  messages: [...state.messages, action.message]
})

/**
 * Reducer
 */
export const reducers = createReducer(INITIAL_STATE, {
  [Types.TOGGLE_CHAT]: toggle,
  [Types.GET_CHAT]: get,
  [Types.ADD_MESSAGE]: addMessage
})
