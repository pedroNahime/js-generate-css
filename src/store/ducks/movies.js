import { createActions, createReducer } from "reduxsauce";

export const { Types, Creators } = createActions({
    getMovie: [],
    showMovie: ["search"],
    successMovie: ["data"],
    errorMovie: ["error"]
});

/**
 * Handlers
 */
const INITIAL_STATE = {
    data: [],
    loading: false,
    search: '',
    error: ''
};

const get = (state = INITIAL_STATE) => ({
        ...state, loading: true
});
const show = (state = INITIAL_STATE, action) => ({
    ...state, action: action.search, loading: true
});
const success = (state = INITIAL_STATE, action) => ({
    ...state, data: action.data, loading: false
});
const error = (state = INITIAL_STATE, action) => ({
    ...state, error: action.error, loading: false
});

/**
 * Reducer
 */
export default createReducer(INITIAL_STATE, {
    [Types.GET_MOVIE]: get,
    [Types.SHOW_MOVIE]: show,
    [Types.SUCCESS_MOVIE]: success,
    [Types.ERROR_MOVIE]: error,
});
