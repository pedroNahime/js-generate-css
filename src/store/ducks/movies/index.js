import { createActions, createReducer } from 'reduxsauce'

export const { Types, Creators } = createActions({
  getMovie: [],
  showMovie: ['search'],
  successMovie: ['data'],
  errorMovie: ['error']
})

export const MoviesTypes = Types
export default Creators
/**
 * Handlers
 */
export const INITIAL_STATE = {
  data: [],
  loading: false,
  search: '',
  error: ''
}

export const get = (state = INITIAL_STATE) => ({
  ...state,
  loading: true
})
export const show = (state = INITIAL_STATE, action) => ({
  ...state,
  search: action.search,
  loading: true
})
export const success = (state = INITIAL_STATE, action) => ({
  ...state,
  data: action.data,
  loading: false
})
export const error = (state = INITIAL_STATE, action) => ({
  ...state,
  error: action.error,
  loading: false
})

/**
 * Reducer
 */
export const reducers = createReducer(INITIAL_STATE, {
  [Types.GET_MOVIE]: get,
  [Types.SHOW_MOVIE]: show,
  [Types.SUCCESS_MOVIE]: success,
  [Types.ERROR_MOVIE]: error
})
