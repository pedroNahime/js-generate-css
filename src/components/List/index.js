import PropTypes from 'prop-types'
import React, { useEffect, useCallback } from 'react'
import { useDispatch } from 'react-redux'
import { bindActionCreators } from 'redux'
import { Creators as MoviesAction } from '../../store/ducks/movies'
import { connect } from 'react-redux'
import Movie from '../Movie'
import { Container, Title, Wrapper } from './styles.js'

const List = props => {
  const dispatch = useDispatch()
  const initFetchMethod = useCallback(() => {
    return dispatch(props.getMovie())
  }, [dispatch])
  useEffect(() => {
    initFetchMethod()
  }, [initFetchMethod])
  return (
    <Container>
      <Title>Movies</Title>
      <Wrapper>
        {props.movies.data.results
          ? props.movies.data.results.map(movie => (
              <Movie key={movie.id} movie={movie} />
            ))
          : null}
      </Wrapper>
    </Container>
  )
}

export const mapStateToProps = state => ({
  movies: state.movies
})

export const mapDispatchToProps = dispatch =>
  bindActionCreators(MoviesAction, dispatch)

export default connect(mapStateToProps, mapDispatchToProps)(List)

List.propTypes = {
  getMovie: PropTypes.func.isRequired,
  movies: PropTypes.object.isRequired
}
