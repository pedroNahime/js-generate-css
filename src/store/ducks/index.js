import { combineReducers } from 'redux'

import { reducers as movies } from './movies'

export default combineReducers({
  movies
})
