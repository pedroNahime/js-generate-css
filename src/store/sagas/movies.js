import {call, put} from 'redux-saga/effects'
import api from '../../services/api'
import {Creators as MoviesActions} from "../ducks/movies";

const key = '';

export function* getMovie() {
    try {
        const {data} = yield call(api.get, `trending/all/day?api_key=${key}`);
        yield put(MoviesActions.successMovie(data))
    } catch (e) {
        const {status_message, status_code} = e.response.data;
        const response = {
            status: e.response.status,
            data: {
                status_message,
                status_code
            }
        };
        yield put(MoviesActions.errorMovie(response))
    }
}