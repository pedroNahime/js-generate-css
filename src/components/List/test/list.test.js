import React from 'react'
import { Creators as MoviesActions } from '../../../store/ducks/movies'
import { render } from '@testing-library/react'
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import { mapDispatchToProps, mapStateToProps } from '../index'
import List from '../index'

describe('<List component/>', () => {
  describe('#mapDispatchToProps', () => {
    let dispach
    let props

    beforeEach(() => {
      dispach = jest.fn()
      props = mapDispatchToProps(dispach)
    })
    test('binds actions', () => {
      props.getMovie()
      expect(dispach).toHaveBeenCalledWith(MoviesActions.getMovie())
    })
  })

  describe('#mapStateToProps', () => {
    let state

    beforeEach(() => {
      state = { movies: [{ id: 1, title: 'someTitle', name: 'someName' }] }
    })

    test('should return state with movies', () => {
      expect(mapStateToProps(state).movies[0].id).toEqual(1)
    })
  })

  test('render component', () => {
    let l = {
      getMovie: jest.fn(),
      movies: {
        data: { results: [{ id: 1, title: 'someTitle', name: 'someName' }] }
      }
    }
    const mockStore = configureStore()
    let store = mockStore(l)
    let { container } = render(
      <Provider store={store}>
        <List />
      </Provider>
    )
    expect(container).toMatchSnapshot()
  })
})
