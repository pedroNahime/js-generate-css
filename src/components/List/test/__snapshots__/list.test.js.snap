// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List component/> render component 1`] = `
<div>
  <div
    class="sc-bwzfXH jawXPz"
  >
    <h1
      class="sc-htpNat gMpIXU"
    >
      Movies
    </h1>
    <div
      class="sc-bxivhb joIOkd"
    >
      <a
        class="sc-bdVaJa jKuiS"
      />
    </div>
  </div>
</div>
`;
